test/integration/warm-pool/warm_pool_suite_test.go

```diff
@@ -0,0 +1,323 @@
+	// scaling occurs, if set above 0 will increment some areas of no cluster scaling (3, 4, 6, 8, 9)
+	maxPods = 60              // max pods you want to work with for your cluster (all)
+	minPods = 0               // tests can be run with a base amount of pods at start (all)
+	sleep   = 1 * time.Minute // sleep interval (all)
+)
+
+// get from k8s client
+var clusterIP = "10.100.140.129" // Get the cluster ip of the prometheus-server service
+var f *framework.Framework
+var err error
+
+// This Results structure defines JSON response from the Prometheus API
+type Result struct {
+	Status string `json:"status"`
+	Data   struct {
+		ResultType string `json:"resultType"`
+		Result     []struct {
+			Metric struct {
+				Name                    string `json:"__name__"`
+				AppKubernetesIoInstance string `json:"app_kubernetes_io_instance"`
+				AppKubernetesIoName     string `json:"app_kubernetes_io_name"`
+				ControllerRevisionHash  string `json:"controller_revision_hash"`
+				Instance                string `json:"instance"`
+				Job                     string `json:"job"`
+				K8SApp                  string `json:"k8s_app"`
+				Namespace               string `json:"namespace"`
+				Node                    string `json:"node"`
+				Pod                     string `json:"pod"`
+				PodTemplateGeneration   string `json:"pod_template_generation"`
+			}
+			Values [][2]interface{} `json:"values"`
+		}
+	}
+}
+
+func TestWarmPool(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "VPC Warm Pool Test Suite")
+}
+
+var _ = BeforeSuite(func() {
+	f = framework.New(framework.GlobalOptions)
+
+	By("creating test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		CreateNamespace(utils.DefaultTestNamespace)
+
+	nodeList, err := f.K8sResourceManagers.NodeManager().GetNodes(f.Options.NgNameLabelKey,
+		f.Options.NgNameLabelVal)
+	Expect(err).ToNot(HaveOccurred())
+
+	numOfNodes := len(nodeList.Items)
+	Expect(numOfNodes).Should(BeNumerically(">", 1))
+})
+
+var _ = AfterSuite(func() {
+	By("deleting test namespace")
+	f.K8sResourceManagers.NamespaceManager().
+		DeleteAndWaitTillNamespaceDeleted(utils.DefaultTestNamespace)
+})
+
+// Helper Functions //
+
+// Basic Prometheus api call
+func callPrometheus(url string) Result {
+	command := []string{"curl", "--silent", "-g", url}
+	stdout, _, err := f.K8sResourceManagers.PodManager().PodExec(utils.DefaultTestNamespace, "curl-pod",
+		command)
+	Expect(err).ToNot(HaveOccurred())
+	Expect(stdout).ShouldNot(BeEmpty())
+	var result Result
+	marshallErr := json.Unmarshal([]byte(stdout), &result)
+	if marshallErr != nil {
+		fmt.Printf("Cannot unmarshall json: %s", marshallErr)
+	}
+	return result
+}
+
+func getPrometheusMetrics(start int64, end int64) {
+	By("Fetching metrics via Curl Container")
+	// metrics
+	warmMetric := "awscni_total_ip_addresses-awscni_assigned_ip_addresses"
+	noAddrsMetric := "awscni_err_no_avail_addrs"
```

Comment:
This needs to be updated after the metrics was renamed to `awscni_no_available_ip_addresses`